craft_application/services/testing.py

```diff
@@ -48,17 +50,26 @@ def process_spread_yaml(self) -> None:
             craft_backend=craft_backend,
         )
 
-        spread_yaml.to_yaml_file(pathlib.Path("spread/spread.yaml"))
+        spread_yaml.to_yaml_file(dest)
+
+    def run_spread(self, project_path: pathlib.Path) -> None:
+        """Run spread on the processed project file.
+
+        :param project_path: The processed project file.
+        """
+        with emit.pause():
+            os.environ["SPREAD_PROJECT_FILE"] = str(project_path)
+            subprocess.run(["spread", "-v", "craft:"], check=True)
 
     def _get_backend(self) -> models.SpreadBackend:
-        name = "lxd-vm"
+        name = "ci" if os.environ.get("CI") else "lxd-vm"
 
         return models.SpreadBackend(
             type="adhoc",
-            allocate=f"ADDRESS $(./.extension allocate {name})",
-            discard=f"./.extension discard {name}",
-            prepare=f'"$PROJECT_PATH"/.extension backend-prepare {name}',
-            restore=f'"$PROJECT_PATH"/.extension backend-restore {name}',
-            prepare_each=f'"$PROJECT_PATH"/.extension backend-prepare-each {name}',
-            restore_each=f'"$PROJECT_PATH"/.extension backend-restore-each {name}',
+            allocate=f"ADDRESS $(./spread/.extension allocate {name})",
```

Comment:
The spread project root has been moved from `spread/` to the craft project root. This allows a project to be built as part of spread testing.